docs/tutorial/part04.rst

```diff
@@ -2,5 +2,321 @@
+
+To start coding, first let's create a new python module under our ``fetch_api/`` directory.
+The next command will create a file named `utils.py`::
+
+    touch fetch_api/utils.py
+
+Now, as we will want to query the Neo4j database, we will import our models.
+Put the below import statements at the start of the file::
+
+    from .models import Entity
+    from .models import Intermediary
+    from .models import Officer
+    from .models import Address
+    from .models import Other
+
+In order to easily access each of the model classes programmatically, let's create a key-value map.
+The key will be the model class name and the value the model class itself::
+
+    MODEL_ENTITIES = {
+        'Entity': Entity,
+        'Address': Address,
+        'Intermediary': Intermediary,
+        'Officer': Officer,
+        'Other': Other
+    }
+
+Filter Nodes Helper
+-------------------
+
+We will create a function that receive a model class and some filter parameters like *name, country
+jurisdiction and source_id*. Then this functin will return a filtered nodeset contaning only the
+model nodes that pass our filters.
+
+Let's call this helper function ``filter_nodes``::
+
+    def filter_nodes(node_type, search_text, country, jurisdiction, source_id):
+        node_set = node_type.nodes
+
+        # On Address nodes we want to check the search_text against the address property
+        # For any other we check against the name property
+        if node_type.__name__ == 'Address':
+            node_set.filter(address__icontains=search_text)
+        else:
+            node_set.filter(name__icontains=search_text)
+
+        # Only entities store jurisdiction info
+        if node_type.__name__ == 'Entity':
+            node_set.filter(jurisdiction__icontains=jurisdiction)
+
+        node_set.filter(countries__icontains=country)
+        node_set.filter(sourceID__icontains=source_id)
+
+        return node_set
+
+Count Nodes Helper
+------------------
+
+We will create a function that return the length of the nodeset returned by the ``filter_nodes`` helper
+we created before. It will receive a dictionary of filters.
+
+Let's call this helper function ``count_nodes``::
+
+    def count_nodes(count_info):
+        count = {}
+        node_type               = count_info['node_type']
+        search_word             = count_info['name']
+        country                 = count_info['country']
+        jurisdiction            = count_info['jurisdiction']
+        data_source             = count_info['sourceID']
+        node_set                = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, jurisdiction, data_source)
+        count['count']          = len(node_set)
+
+        return count
+
+Fetch Nodes Helper
+------------------
+
+We will create a function that return a subset of nodes from the nodeset returned by the
+``filter_nodes`` helper we created before. It will receive a dictionary of filters.
+

```

Comment:
you use return twice in the same sentence, maybe 
`We will create a function that returns a subset of nodes  filtered by the `` filter_nodes`` method that we created previously`